### PR TITLE
[serve] Update replica ID format (and other random strings) to better match UUIDs

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

Small cosmetic update to our replica ID generation to make it use the same alphabet as `UUID` (which is standard and we use for request IDs).

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/_private/controller.py [hunk 1]
```diff
@@ -614,7 +614,7 @@ def shutdown(self):
             and proxy_state_is_shutdown
         ):
             logger.warning(
-                "All resources have shut down, shutting down controller!",
```
Comment (edoakes) [inline python/ray/serve/_private/controller.py:617]: @GeneDer reminder to mostly avoid exclamation points in log messages (and other technical writing)

Comment (GeneDer) [reply python/ray/serve/_private/controller.py:617]: will do👍

**Review by GeneDer — APPROVED** [review]
:shipit:

File: python/ray/serve/_private/utils.py [hunk 2]
```diff
@@ -134,8 +134,11 @@ def block_until_http_ready(
         time.sleep(backoff_time_s)
 
 
-def get_random_letters(length=6):
-    return "".join(random.choices(string.ascii_letters, k=length))
+# Match the standard alphabet used for UUIDs.
+RANDOM_STRING_ALPHABET = string.ascii_lowercase + string.digits
```
[the remark this answers is not in the document]
Comment (GeneDer) [reply python/ray/serve/_private/utils.py:138]: Also kinda after thought, there is also a [UUID](https://docs.python.org/3/library/uuid.html#module-uuid) module in python not sure if we want to consider using that

[the remark this answers is not in the document]
Comment (edoakes) [reply python/ray/serve/_private/utils.py:138]: yes we use `UUID` elsewhere but here the length of the UUID is overkill and will degrade the UX for reading logs (replica IDs are scoped to a single deployment in a single cluster)

**Review by sihanwang41 — APPROVED** [review]

**Review by shrekris-anyscale — APPROVED** [review]

**Pull request merged**